satpy/readers/atms_l1b_nc.py

```diff
@@ -0,0 +1,153 @@
+
+    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+        """Initialize file handler."""
+        super().__init__(
+            filename, filename_info, filetype_info, auto_maskandscale=True,
+        )
+        self.channel_name_to_index = ATMS_CHANNEL_NAME_TO_INDEX
+
+    @property
+    def start_time(self):
+        """Get observation start time."""
+        return datetime.strptime(self['/attr/time_coverage_start'], DATE_FMT)
+
+    @property
+    def end_time(self):
+        """Get observation end time."""
+        return datetime.strptime(self['/attr/time_coverage_end'], DATE_FMT)
+
+    @property
+    def platform_name(self):
+        """Get platform name."""
+        return self["/attr/platform"]
+
+    @property
+    def sensor(self):
+        """Get sensor."""
+        return self["/attr/instrument"]
+
+    @property
+    def antenna_temperature(self):
+        """Get antenna temperature."""
+        file_key = self.filetype_info["antenna_temperature"]
+        return self[file_key]
+
+    @property
+    def attrs(self):
+        """Return attributes."""
+        return {
+            "filename": self.filename,
+            "start_time": self.start_time,
+            "end_time": self.end_time,
+            "platform_name": self.platform_name,
+            "sensor": self.sensor,
+        }
+
+    @staticmethod
+    def _standardize_dims(dataset):
+        """Standardize dims to y, x."""
+        if "atrack" in dataset.dims:
+            dataset = dataset.rename({"atrack": "y"})
+        if "xtrack" in dataset.dims:
+            dataset = dataset.rename({"xtrack": "x"})
+        if dataset.dims[0] == "x":
+            dataset = dataset.transpose("y", "x")
+        return dataset
+
+    @staticmethod
+    def _drop_coords(dataset):
+        """Drop coords that are not in dims."""
+        for coord in dataset.coords:
+            if coord not in dataset.dims:
+                dataset = dataset.drop_vars(coord)
+        return dataset
+
+    def _merge_attributes(self, dataset, dataset_info):
+        """Merge attributes of the dataset."""
+        dataset.attrs.update(self.filename_info)
+        dataset.attrs.update(dataset_info)
+        dataset.attrs.update(self.attrs)
+        return dataset
+
+    def _get_channel_data(self, channel_name):
+        """Get channel data."""
+        idx = self.channel_name_to_index.get(channel_name)
+        return self.antenna_temperature[:, :, idx]
+
+    def get_dataset(self, dataset_id, ds_info):
+        """Get dataset using file_key in dataset_info."""
+        param = dataset_id['name']
+        logger.debug(f'Reading in file to get dataset with key {param}.')
+        try:
+            dataset = (
+                self[param] if param not in self.channel_name_to_index
+                else self._get_channel_data(param)
+            )
+        except KeyError:
+            logger.warning(f'Could not find {param} in NetCDF file, no valid Dataset created')  # noqa: E501
```

Comment:
Is this try/except needed? Is the error message that is generated by default (without this try/except) too confusing? I'm not sure this adds much to the log. Additionally, I'm not sure returning `None` is the expected behavior when a dataset can't be loaded. I think `KeyError` should be handled just fine.

Reply:
I checked and found out this try/except is not needed. The default error message is good enough,  so I removed it.